src/main/java/com/booksaw/betterTeams/ConfigManager.java

```diff
@@ -102,26 +104,34 @@ public void updateFromDefaultSave(boolean log) {
 		}
 
 		List<String> changes = updateFileConfig(Main.plugin.getResource(resourceName));
+		int migratedKeys = updateVariables();
 
 		if (log) {
-			if (changes.isEmpty()) {
+			if (changes.isEmpty() && migratedKeys == 0) {
```

Comment:
With this feature it may have cause cleaner code (with the other changes suggested to do something like)
```java
if (!changes.isEmpty) {
 // log changes
}
if (changes.isEmpty() && !migratedKeys) {
 // log that nothing has changed
} else {
// save config
}
```

As a general practice (not one that I have even always followed myself when writing BetterTeams) avoiding creating long nests of if statements makes the code more maintainable and less complex for a reader.